src/ScriptBuilder/Saga/SagaDefinitionReader.cs

```diff
@@ -38,6 +45,192 @@ public static bool TryGetSqlSagaDefinition(TypeDefinition type, out SagaDefiniti
         return true;
     }
 
+    static bool TryInferSagaDefinition(TypeDefinition type, out SagaDefinition definition)
+    {
+        var baseType = type.BaseType as GenericInstanceType;
+        definition = null;
+
+        // Class must directly inherit from NServiceBus.Saga<T> so that no tricks can be pulled from an intermediate class
+        if (baseType == null || !baseType.FullName.StartsWith("NServiceBus.Saga") || baseType.GenericArguments.Count != 1)
```

Comment:
> !baseType.FullName.StartsWith("NServiceBus.Saga")

given namespaces reshuffling etc. should we add an explicit test that this assumption is correct for all (future) versions of core?

Reply:
> baseType.GenericArguments.Count != 1

is that check necessary if we're already checking the name?

Reply:
The SqlP package is only scoped to work with CoreV7 (or for the backport, CoreV6) and that API can't go away, so I don't know what else we should/could test?

The generic arguments count is necessary because there's `NServiceBus.Saga<T>` which we want, and then there's `NServiceBus.Saga` (no generics) which is the base class of `Saga<T>`, which we don't want.